lib/Table/TableColumn.ts

```diff
@@ -97,6 +100,91 @@ export default class TableColumn {
     return result;
   }
 
+  /**
+   * Gets `math-expression-evaluator` tokens. These allow other column values to be used in an expression.
+   * Each token is generated by `transformation.dependencies` trait - which is an array of strings of column names.
+   * A token `x` is also added - which corresponds to the value in **this column**.
+   * For example:
+   * - if `dependencies = ['columnA']`
+   * - then a token will be generated for `columnA`
+   * - and then we can access the value in an expression (eg `x*columnA` will multiple each value in this column, by the value in `columnA`)
+   */
+  @computed
+  get mexpColumnTokens() {
+    return [
+      {
+        type: 3,
+        token: "x",
+        show: "x",
+        value: "x"
+      },
+      ...filterOutUndefined(
+        this.traits.transformation?.dependencies?.map(colName => {
+          if (this.tableModel.tableColumns.find(col => col.name === colName)) {
+            return {
+              type: 3,
+              token: colName,
+              show: colName,
+              value: colName
+            };
+          }
+        }) ?? []
+      )
+    ];
+  }
+
+  /**
+   * Gets a function which can be used to generate `math-expression-evaluator` pairs for a given row `i`.
+   * The function returns key-value pairs which map tokens to values.
+   * Each token (see `this.mexpColumnTokens`) represents another column in the table, each value represents corresponding cell for row i.
+   * Note: this does not add pair for token `x` (which corresponds to values in this column), this is in `valuesAsNumbers` immediately before expression is evaluated
+   *
+   * @param i row number
+   * @param x value to transform
+   */
+  mexpColumnPairs(i: number, x: number) {
+    return this.mexpColumnTokens.reduce<{ [key: string]: number | null }>(
+      (pairs, token) => {
+        if (token.token !== "x")
+          pairs[token.value] =
+            this.tableModel.tableColumns.find(col => col.name === token.token)
+              ?.valuesAsNumbers.values[i] ?? null;
+        return pairs;
+      },
+      { x } // Add column pair for this value (token "x")
+    );
+  }
+
+  /**
+   * Gets `math-expression-evaluator` expression in "postfix" notation. This is used by `this.valuesAsNumbers` to transform values.
```

Comment:
I don't really know what to add here, it is how the library works

Reply:
"Gets `math-expression-evaluator` expression in "postfix" notation, which is necessary for `math-expression-evaluator` to evaluate it."

I know it's how the library works, but adding a bit more detail helps people reading the code orient themselves. It makes sense to you, because you know the library, but if someone's going through this for the first time, they won't know what they're looking at. It took me a solid 40 minutes to parse this PR! which is the way it is sometimes, but the easier you can make it, the better.

Reply:
I'll rearrange the comments to make it more obvious